active/XamlServiceProvider/XamlServiceProviderApi.md

```diff
@@ -0,0 +1,272 @@
+
+The following example shows a custom markup extension that returns a string describing the object at the root of the markup. It does this by getting the IRootObjectProvider from the IXamlServiceProvider parameter passed to ProvideValue.
+
+With this markup extension:
+
+```cs
+public class TestMarkupExtension : MarkupExtension
+{
+    public override object ProvideValue(IXamlServiceProvider serviceProvider)
+    {
+        var target = serviceProvider.GetService(typeof(IRootObjectProvider)) as IRootObjectProvider;
+        return target.RootObject.ToString();
+    }
+}
+```
+
+the TextBlock in this markup will display “App1.MainPage”:
+
+```xml
+<Page x:Class="App1.MainPage"
+    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+    xmlns:local="using:App52"
+    Background="{ThemeResource ApplicationPageBackgroundThemeBrush}" >
+
+    <Grid>
+        <TextBlock Text="{local:TestMarkupExtension}" />
+    </Grid>
+</Page>
+```
+
+## IUriContext 
+
+The following example shows a custom markup extension that converts a relative URI to an absolute URI. It does this by getting the IUriContext from the IXamlServiceProvider parameter passed in to ProvideValue. IUriContext provides the URI of the markup being loaded.
+
+This markup extension enables the following markup, where a property requires an absolute URI:
+
+```xml
+<SomeControl AbsoluteUri="{RelativeUriResolver 'image.jpg'}"
+```
+
+```cs
+public class RelativeUriResolver : MarkupExtension
+{
+    public RelativeUriResolver(Uri uri)
+    {
+        SourceUri = uri;
+    }
+    public Uri SourceUri { get; set; }
+    protected override object ProvideValue(IXamlServiceProvider provider)
+    {
+        if(SourceUri == null || SourceUri.IsAbsoluteUri)
+        {
+            return SourceUri;
+        }
+
+        var uriContext = provider.GetService(typeof(IUriContext)) as IUriContext;
+        if (uriContext != null || uriContext.BaseUri != null)
+        {
+            // For example this will calculate "ms-appx:///Controls/ColorPicker.xaml" + "colorImage.jpg" to 
+            // "ms-appx:///Controls/colorImage.jpg"
+            if (Uri.TryCreate(uriContext.BaseUri, SourceUri, out var absoluteUri))
+            {
+                return absoluteUri;
+            }
+        }
+
+        return SourceUri;
+    }
+}
+```
+
+# API Notes/Remarks
+
+## MarkupExtension.ProvideValue(IXamlServiceProvider) virtual method
+
+When implemented in a derived class, returns an object that is provided as the value of the target property for this markup extension.
+
+**Remarks**
+
+The IXamlServiceProvider argument can be used to retrieve the following services:
+* `IProvideValueTarget`, which provides information about the type/property the markup extension is being applied to. This aligns with WPF's [IProvideValueTarget](https://docs.microsoft.com/dotnet/api/System.Windows.Markup.IProvideValueTarget).

```

Comment:
Lines 190-196 seem like they're on the wrong place? Shouldn't they be under the `IXamlServiceProvider` section?

Reply:
As for where the interface belongs, we'll follow up with the above-mentioned cs/winrt issue